src/main/python/run_trecqa.py

```diff
@@ -5,219 +5,294 @@
-        jaccard_similarity = value[1]
-        i += 1
-
-        # check if the answer already exists in the ranked-list
-        if jaccard_similarity >= threshold:
-          doc_id = value[0][0]
-          if doc_id not in seen_docs:
-            out.write('{} Q0 {} {} {} TRECQA\n'.format(qid, doc_id, i, value[2]))
-            labels_predicted[qid].append(doc_id)
-            seen_docs.append(doc_id)
+    parser = argparse.ArgumentParser(description='Evaluate the QA system')
+    parser.add_argument('-input', help='path of a TrecQA file', required=True)
+    parser.add_argument("-output", help="path of output file", default=".")
+    parser.add_argument("-qrel", help="path of qrel file")
+    parser.add_argument("-h0", help="number of hits", default=1000)
+    parser.add_argument("-h1", help="h1 passages to be reranked", default=100)
+    parser.add_argument("-k", help="evaluate at depth k", default="5")
+    parser.add_argument("-model", help="[idf|sm]", default='idf')
+    parser.add_argument('-index', help="path of the index", required=True)
+    parser.add_argument('-w2v-cache', help="word embeddings cache file")
+    parser.add_argument('-qa-model-file', help="the path to the model file")
+    parser.add_argument('-eval', help="[pattern|jaccard]", default='pattern')
+    parser.add_argument('-pattern', help='path to the pattern file')
+
+    args = parser.parse_args()
+
+    if (args.model == "sm" and not args.w2v_cache and not args.qa_model_file):
+        print("Pass the word embeddings cache file and the model file")
+        parser.print_help()
+        exit()
+
+    pattern_dict = {}
+    if args.eval == 'pattern':
+        if not args.pattern:
+            print("Path to the pattern file should be included if the method of evaluation is pattern based")
+            parser.print_help()
+            exit()
         else:
-          unjudged_count += 1
-          doc_id = 'unjudged{}'.format(unjudged_count)
-          out.write('{} Q0 {} {} {} TRECQA\n'.format(qid, doc_id, i, value[2]))
-          labels_predicted[qid].append(doc_id)
-
-  depth_range = [5, 10, 15, 20, 50, 100]
-  for k in depth_range:
-    print('P@{}: {}'.format(k, get_precision(labels_actual, labels_predicted, k)))
+            with open(args.pattern, 'r') as pattern_file:
+                for line in pattern_file:
+                    det = line.strip().split(' ', 1)
+
+                    if det[0] not in pattern_dict:
+                        pattern_dict[det[0]] = det[1]
+
+    pyserini = Pyserini(args.index)
+    questions, answers, labels_actual = load_data(args.input)
+    threshold = 0.7
+    # TODO: what is this ^^ magic number
+
+    qrel_file = ""
+
+    if args.qrel:
+        qrel_file = args.qrel
+    elif args.eval == 'jaccard':
+        qrel_file = '{}/qrel.jaccard.txt'.format(args.output)
+        create_qrel_jaccard(qrel_file, answers)
+    elif args.eval == 'pattern':
+        qrel_file = '{}/qrel.pattern.txt'.format(args.output)
```

Comment:
Double check that the description of `args.output` in the earlier file matches how it's used here -- my recollection is that it isn't.